rep-2001.rst

```diff
@@ -33,15 +33,63 @@ Specification
 End-user entry points
 ---------------------
 
-We define three main entry points for ROS users.
+We define these main entry points for ROS users.
 
  * desktop (recommended)
  * ros_base
  * ros_core
+ * ros_cpp_core (advanced)
 
 Variants
 --------
 
+ROS C++ Core
+'''''''''''
+
+The `ros_cpp_core` variant is a restricted subset of `ros_core` functionality with support for only C and C++.
+It targets resource- and tool-constrained environments where package managers may be limited or unavailable.
+The intention is that this variant can target cross-compiling for custom OS sysroots that may not have build tools or Python available that run on the target platform.
+Build tools may use Python, this variant is not intended to restrict build tooling.
+It may not contain any GUI or Python dependencies.
+It may not have any dependencies outside a source workspace of its members.
+Note that this variant chooses an RMW and rcl_logging implementation based on their minimal dependency footprint.
+
+::
+
+ - ros_cpp_core:
+      # These are the packages we are actually targeting
+      packages_up_to: [rclcpp, rclcpp_action, rclcpp_lifecycle, rclcpp_components]
+      # This is the full package set that is actually built, and needs to meet above specified dependency criteria
+      packages_all: [
+            action_msgs, ament_cmake, ament_cmake_core,
+            ament_cmake_export_definitions, ament_cmake_export_dependencies,
+            ament_cmake_export_include_directories,
+            ament_cmake_export_interfaces, ament_cmake_export_libraries,
+            ament_cmake_export_link_flags, ament_cmake_gmock,
+            ament_cmake_gtest, ament_cmake_include_directories,
+            ament_cmake_libraries, ament_cmake_pytest, ament_cmake_python,
+            ament_cmake_ros, ament_cmake_target_dependencies, ament_cmake_test,
+            ament_cmake_version, ament_index_cpp, ament_index_python,
+            ament_package, builtin_interfaces, class_loader,
+            composition_interfaces, console_bridge_vendor, cyclonedds,
+            cyclonedds_cmake_module, domain_coordinator, libyaml_vendor,
```

Comment:
nit: Maybe we should use the same convention like other variants to describe the DDS vendor dependency.

```
 And at least one of the following rmw_implementation:
     - Fast-RTPS: [Fast-CDR, Fast-RTPS, rmw_fastrtps]
     - Connext: [rmw_connext, rosidl_typesupport_connext]
     - Opensplice: [rmw_opensplice, rosidl_typesupport_opensplice]
```

Reply:
Please see the above conversation why only CycloneDDS is listed here: https://github.com/ros-infrastructure/rep/pull/231#discussion_r374851508